src/libCps/ScillaHelpersCall.cpp

```diff
@@ -219,15 +228,32 @@ ScillaCallParseResult ScillaHelpersCall::ParseCallContractJsonOutput(
 
     // ZIL-5165: Don't fail if the recipient is a user account.
     {
-      const CpsAccountStoreInterface::AccountType accountType = acc_store.GetAccountType(recipient);
+      const CpsAccountStoreInterface::AccountType accountType =
+          acc_store.GetAccountType(recipient);
       LOG_GENERAL(INFO, "Target is accountType " << accountType);
       if (accountType == CpsAccountStoreInterface::DoesNotExist ||
           accountType == CpsAccountStoreInterface::EOA) {
         LOG_GENERAL(INFO, "Target is EOA: processing.");
-        // Message sent to a non-contract account. Add something to results.entries so that if this
-        // message attempts to transfer funds, it succeeds.
-        results.entries.emplace_back(ScillaCallParseResult::SingleResult{
-            {}, recipient, amount, false});
+        // Message sent to a non-contract account. Add something to
+        // results.entries so that if this message attempts to transfer funds,
+        // it succeeds.
+        results.entries.emplace_back(
+            ScillaCallParseResult::SingleResult{{}, recipient, amount, false});
+        continue;
+      }
+    }
+
+    if (acc_store.isAccountEvmContract(recipient)) {
+      // Workaround before we have full interop: treat EVM contracts as EOA
+      // accounts only if there's receiver_address set to 0x0, otherwise revert
+      if (!scillaArgs.extras ||
```

Comment:
I think this has the behaviour that:
 - if you call Scilla from EVM with receiver_address = 0
 - and some contract down the chain sends a message 
 -  and that message is to an EVM contract
 - and that EVM contract has a handler for the message
 - then we ignore the handler and do nothing.

Which isn't what RFC075 says .. is there a test for this that this code passes?

Reply:
Nope, that should be reverted in this case because receiver_address will be missing or (impossible with current impl) be non-zero.